hack/verify-no-vendor-cycles.sh

```diff
@@ -18,13 +18,24 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+
+staging_repos=($(ls "${KUBE_ROOT}/staging/src/k8s.io/"))
+staging_repos_pattern=$(IFS="|"; echo "${staging_repos[*]}")
+
 failed=false
 for i in $(find vendor/ -type d); do
-  deps=$(go list -f '{{range .Deps}}{{.}}{{"\n"}}{{end}}' ./$i 2> /dev/null | grep -v "k8s.io/kubernetes/vendor/" | grep "k8s.io/kubernetes" || echo "")
-    if [ -n "${deps}" ]; then
+  deps=$(go list -f '{{range .Deps}}{{.}}{{"\n"}}{{end}}' ./$i 2> /dev/null || echo "")
+  deps_on_main=$(echo "${deps}" | grep -v "k8s.io/kubernetes/vendor/" | grep "k8s.io/kubernetes" || echo "")
+  if [ -n "${deps_on_main}" ]; then
     echo "Package ${i} has a cyclic dependency on the main repository."
     failed=true
   fi
+  deps_on_staging=$(echo "${deps}" | grep "k8s.io/kubernetes/vendor/k8s.io" | grep -E ${staging_repos_pattern} || echo "")
```

Comment:
Is this going to flag the staging repos themselves? (since they are symlinked into vendor/)